plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoPrivilegesCaseInsensitiveSchema.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.mongodb;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoDatabase;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.MaterializedRow;
+import io.trino.testing.QueryRunner;
+import org.bson.Document;
+import org.testng.annotations.Test;
+
+import java.util.Locale;
+import java.util.Optional;
+
+import static io.airlift.testing.Closeables.closeAllSuppress;
+import static io.trino.plugin.mongodb.AuthenticatedMongoServer.createRole;
+import static io.trino.plugin.mongodb.AuthenticatedMongoServer.createUser;
+import static io.trino.plugin.mongodb.AuthenticatedMongoServer.privilege;
+import static io.trino.plugin.mongodb.AuthenticatedMongoServer.resource;
+import static io.trino.plugin.mongodb.AuthenticatedMongoServer.role;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestMongoPrivilegesCaseInsensitiveSchema
```

Comment:
Let's add new test cases to `TestMongoCaseInsensitiveMapping`

Reply:
I'd be happy to, but this test is almost identical `TestMongoPrivileges` from which it was copy-pasted.
If I understood correctly such complex structure is due to dependency of the trino configuration property `mongodb.connection-url` from `mongoServer.testUserConnectionString()` where `TEST_USER` and `TEST_PASSWORD` are passed. 
May be @huberty89 @ebyhr know more. 
Also @kokosing may be the solution is much easier than I think, so I'd be happy to implement it if you advice.

Reply:
Hmm, I'm not sure if I understand the problem, could you create query runner with two mongo connectors configured - everyone with different user?

Reply:
I've merged my implementation with TestMongoPrivileges. 
However merging it with `TestMongoCaseInsensitiveMapping` is not a trivial task. Otherwise I guess `TestMongoPrivileges` would never appear.